Dynatrace/dynatrace-operator
# Only convert metadata enrichment if oneagent injection was necessary

## Description


~This small change in conversion logic was discussed last week, but it seems to have been forgotten.~
~- Or we decided against it, but then I wasn't informed, and we can just close it 😅~

As it turns out we need to be even stricter and only convert metadata enrichment if oneagent injection was necessary in the original dynakube


## How can this be tested?


Deploy a classicFullStack/hostmonitoring/activegate-only v1beta1 DynaKube, and see what happens 😉
